.github/workflows/sp-deployment-pipeline.yml

```diff
@@ -0,0 +1,56 @@
+name: Deploy Sample Platform
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ master ]
+
+# permission can be added at job level or workflow level
+permissions:
+  id-token: write
+  contents: read # This is required for actions/checkout
+
+env:
+  DEPLOY_BRANCH: main
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_deployment:
```

Comment:
I think `workflow_dispatch` at the top lets us run the Action manually. I don't understand the purpose of this `workflow_deployment`.